miden-tx/src/testing/data_store.rs

```diff
@@ -89,29 +75,15 @@ impl DataStore for MockDataStore {
         block_num: u32,
         notes: &[NoteId],
     ) -> Result<TransactionInputs, DataStoreError> {
-        assert_eq!(account_id, self.account.id());
-        assert_eq!(block_num, self.block_header.block_num());
-        assert_eq!(notes.len(), self.notes.len());
-
-        let notes = self
-            .notes
-            .iter()
-            .filter(|note| notes.contains(&note.id()))
-            .cloned()
-            .collect::<Vec<_>>();
+        assert_eq!(account_id, self.tx_inputs.account().id());
+        assert_eq!(block_num, self.block_header().block_num());
+        assert_eq!(notes.len(), self.tx_inputs.input_notes().num_notes());
```

Comment:
we should probably assert that all the requested notes are contained within `self.tx_inputs.input_notes()`.